spacepy/pybats/kyoto.py

```diff
@@ -234,41 +234,38 @@ def add_histplot(self, target=False, loc=111, label='Kyoto $K_{p}$',
             two.  Defaults to **None**, meaning that the full time range
             available is used.
 
+        filled : Boolean
+            If True, make a filled 'traffic light' plot of Kp using
+            spacepy.plot.levelPlot. Extra keyword arguments for levelPlot can
+            be supplied via level_kwargs.
+
         Extra keyword arguments are passed to :function:`matplotlib.pyplot.plot`
-        to customize the line style.
+        to customize the line style (and are ignored if filled is True).
 
         Examples
         ========
+        >>> import matplotlib.pyplot as plt
         >>> import spacepy.pybats.kyoto as kt
         >>> kp = kt.fetch('kp', (1981, 11), (1981, 11)
-        >>> kp.add_histplot(lw=2.0, lc='r', label='Example KP')
+        >>> kp.add_histplot(lw=2.0, color='r', label='Example Kp')
+        >>> ax = plt.gca()
+        >>> kp.add_histplot(filled=True)
 
         '''
-
         import matplotlib.pyplot as plt
 
-        # Shortcuts for the lazy.
-        bstart=self['binstart']
-        bstop =self['binstop']
-        npts  =self.attrs['npts']
-
-        # Reformulate time to get histogram-type look.
-        newtime=np.zeros(npts*24, dtype=object)
-        newkp  =np.zeros(npts*24)
-
-        # Set time range.
         if not time_range:
-            time_range = newtime
-        
-        for i in range(npts*8):
-            newtime[3*i  ] = bstart[i]
-            newtime[3*i+1] = self['time'][i]
-            newtime[3*i+2] = bstop[i]
-            newkp[3*i:3*i+3] = self['kp'][i], self['kp'][i], self['kp'][i]
+            time_range = self['time']
     
-
-        fig, ax = set_target(target, figsize=(10,4), loc=loc)
-        line=ax.plot(newtime, newkp, label=label, **kwargs)
+        fig, ax = set_target(target, figsize=(10,4),  loc=loc)
+        if not filled:
+            tvar = self['binstart'].tolist()
+            tvar.append(tvar[-1]+dt.timedelta(hours=3))
+            kpvar = self['kp'].tolist()
+            kpvar.append(kpvar[-1])
+            line = ax.plot(tvar, kpvar, label=label, ds='steps-post', **kwargs)
```

Comment:
```suggestion
            line = ax.plot(tvar, kpvar, label=label, drawstyle='steps-post',
            **kwargs)
```

Reply:
`ds` was added in matplotlib 3.1.0 (and then a linebreak to keep under 80 columns).